src/cmd/graphql.go

```diff
@@ -0,0 +1,213 @@
+		headers := map[string]string{}
+		for _, value := range headersValue {
+			matches := keyValueExp.FindStringSubmatch(value)
+			headers[matches[1]] = matches[2]
+		}
+
+		cobra.CheckErr(err)
+		paginate, err := flags.GetBool("paginate")
+		handleErr("error getting paginate flag", err)
+
+		aggregate, err := flags.GetString("aggregate")
+		jq, err := gojq.Parse(aggregate)
+		handleErr("error parsing pagination flag value", err)
+		aggregation, err := gojq.Compile(jq)
+
+		handleErr("error compiling pagination flag value", err)
+		queryValue, err := flags.GetString("query")
+		handleErr("error getting query flag value", err)
+		queryParsed, err := convert(queryValue)
+		query, ok := queryParsed.(string)
+		if !ok {
+			handleErr("error parsing query flag value", fmt.Errorf("'%#v' is not a string", queryParsed))
+		}
+		operationName, err := flags.GetString("operationName")
+		cobra.CheckErr(err)
+		fields, err := flags.GetStringArray("field")
+		handleErr("error getting field flag value", err)
+
+		variables := map[string]interface{}{}
+		for _, field := range fields {
+			matches := keyValueExp.FindStringSubmatch(field)
+			value, err := convert(matches[2])
+			handleErr(fmt.Sprintf("error parsing variable '%s'", field), err)
+			variables[matches[1]] = value
+		}
+
+		client := getClientGQL(opslevel.SetHeaders(headers))
+		var output []interface{}
+
+		hasNextPage := true
+		for hasNextPage {
+			data, err := client.ExecRaw(query, variables, opslevel.WithName(operationName))
+			handleErr("error making graphql api call", err)
+			for _, item := range handleAggregate(data, aggregation) {
+				output = append(output, item)
+			}
+
+			if paginate {
+				hasNextPage, err = strconv.ParseBool(string(hasNextPageExp.FindSubmatch(data)[1]))
+				handleErr("error parsing bool for has next page", err)
+				variables["endCursor"] = string(endCursorExp.FindSubmatch(data)[1])
+			} else {
+				hasNextPage = false
+			}
+		}
+
+		json, err := json.Marshal(output)
+		handleErr("error marshaling output to json", err)
+
+		fmt.Println(string(json))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(graphqlCmd)
+
+	graphqlCmd.Flags().StringArrayP("header", "H", nil, "Add a HTTP request header in `key=value` format")
+	graphqlCmd.Flags().BoolP("paginate", "p", false, "Automatically make additional requests to fetch all pages of results")
+	graphqlCmd.Flags().StringP("aggregate", "a", ".", "JQ expression to use to aggregate results")
+	graphqlCmd.Flags().StringP("query", "q", "", "The query or mutation body to use")
+	graphqlCmd.Flags().StringP("operationName", "o", "Raw", "The query or mutation 'operation name' to use")
+	graphqlCmd.Flags().StringArrayP("field", "f", nil, "Add a variable in `key=value` format")
+}
+
+func handleErr(msg string, err error) {
+	if err != nil {
+		log.Error().Err(err).Msg(msg)
+		os.Exit(1)
+	}
+}
+
+func convert(v string) (interface{}, error) {
+	if v == "-" {
+		reader := bufio.NewReader(os.Stdin)
+		return reader.ReadString('\n')
```

Comment:
recurse here on convert so that if the stdin or a file are read in they still convert to a int or bool